docs/wiki/TestingDesign.md

```diff
@@ -84,108 +83,90 @@ Currently the list of the parameter file used to test the module is hardcoded in
-  - NetworkWatcher (needs to reside in same RG)
-  - NetworkSecurityGroups
-  - StorageAccount (Diagnostics)
-- SQLManagedInstances
-  - VirtualNetwork / Subnet
-  - KeyVault / Key
-  - StorageAccount / vulnerabilityAssessmentsStorageAccountId
-- SQLManagedInstancesDatabases
-  - SQLManagedInstances
-- VirtualMachines
-  - VirtualNetwork / Subnet
-  - KeyVault / Secrets (admiUsername, adminPassword)
-- VirtualMachineScaleSet
-  - VirtualNetwork / Subnet
-  - KeyVault / Secrets (admiUsername, adminPassword)
-- VirtualNetworkGateway
-  - PublicIPPrefix
-  - VirtualNetwork (in the same rg, like AFW)
-- VirtualNetworkGatewayConnection
-  - KeyVault / Secret (vpnSharedKey)
-- WebApp
-  - AppServicePlan
-- AvdApplicationGroups
-  - AvdHostPool
-- AvdApplications
-  - AvdAppliccationGroups
-
-#### Services (in order)
-
-Provided the dependencies listed above, the following list of services need to be deployed to be able to test all the modules:
-
-1. VirtualNetwork
-1. StorageAccounts
-1. KeyVault
-1. LogAnalytics
-1. PublicIPPrefix
-1. PublicIPAddresses
-1. EventHubNamespaces
-1. EventHubs (only needed if EventHubs are used for Monitoring in addition to LogAnalytics)
-1. ActionGroup
-1. AzureSqlServer
-1. AppServicePlan
-1. SharedImageGallery
-1. ApplicationSecurityGroups
-1. NetworkSecurityGroups
-1. WvdHostPool
-1. WvdAppliccationGroups
-1. Managed Service Identity
-1. Deployment Scripts
+If we speak from **modules** in this context we mean the **services** which get created from these modules.
+
+#### Resources deployed by the dependency workflow
+
+Together with the resource modules workflows, we are providing a dependency workflow, '_.github\workflows\platform.dependencies.yml_', leveraging resource parameters from the '_utilities\dependencies_' subfolder.
+
+The resources deployed by the dependency workflow need to be in place before testing all the modules. Some of them (e.g. storage account, key vault and event hub namespace) require a globally unique resource name. Before running the dependency workflow, it is required to update those values and the corresponding references in the resource modules parameters.
+
+Since also dependency resources are in turn subject to dependencies with each others, resources are deployed in the following group order.
+
+**First level resources**
+
+  1. Resource Groups: Leveraged by all modules. Multiple instances are deployed:
+     - 'validation-rg': The resource group to which resources are deployed by default during the test deployment phase. This same resource group is also the one hosting the dependencies.
+     - 'artifacts-rg': The resource group to which template specs are published during the publishing phase.
+
+**Second level resources**: This group of resources has a dependency only on the resource group which will host them. Resources in this group can be deployed in parallel.
+
+  2. User assigned identity: This resource is leveraged as a test identity by all resources supporting RBAC.
```

Comment:
I'd probably suggest to restart with `1` for each group.

Reply:
done